batavia/modules/math.js

```diff
@@ -0,0 +1,201 @@
+        throw new batavia.builtins.NotImplementedError("math.degrees has not been implemented");
+    },
+
+    erf: function() {
+        throw new batavia.builtins.NotImplementedError("math.erf has not been implemented");
+    },
+
+    erfc: function() {
+        throw new batavia.builtins.NotImplementedError("math.erfc has not been implemented");
+    },
+
+    exp: function(x) {
+        return new batavia.types.Float(Math.exp(x.__float__().val));
+    },
+
+    // TODO: this isn't supported in PhantomJS
+    expm1: function(x) {
+        return new batavia.types.Float(Math.expm1(x.__float__().val));
+    },
+
+    fabs: function() {
+        throw new batavia.builtins.NotImplementedError("math.fabs has not been implemented");
+    },
+
+    factorial: function() {
+        throw new batavia.builtins.NotImplementedError("math.factorial has not been implemented");
+    },
+
+    floor: function(x) {
+        return new batavia.types.Int(Math.floor(x.__float__().val));
+    },
+
+    fmod: function() {
+        throw new batavia.builtins.NotImplementedError("math.fmod has not been implemented");
+    },
+
+    frexp: function() {
+        throw new batavia.builtins.NotImplementedError("math.frexp has not been implemented");
+    },
+
+    fsum: function() {
+        throw new batavia.builtins.NotImplementedError("math.fsum has not been implemented");
+    },
+
+    gamma: function() {
+        throw new batavia.builtins.NotImplementedError("math.gamma has not been implemented");
+    },
+
+    gcd: function() {
+        throw new batavia.builtins.NotImplementedError("math.gcd has not been implemented");
+    },
+
+    // TODO: this isn't supported in PhantomJS
+    hypot: function(x, y) {
+        return new batavia.types.Float(Math.hypot(x.__float__().val, y.__float__().val));
+    },
+
+    isclose: function() {
+        throw new batavia.builtins.NotImplementedError("math.isclose has not been implemented");
+    },
+
+    isfinite: function() {
+        throw new batavia.builtins.NotImplementedError("math.isfinite has not been implemented");
+    },
+
+    isinf: function() {
+        throw new batavia.builtins.NotImplementedError("math.isinf has not been implemented");
+    },
+
+    isnan: function() {
+        throw new batavia.builtins.NotImplementedError("math.isnan has not been implemented");
+    },
+
+    ldexp: function() {
+        throw new batavia.builtins.NotImplementedError("math.ldexp has not been implemented");
+    },
+
+    lgamma: function() {
+        throw new batavia.builtins.NotImplementedError("math.lgamma has not been implemented");
+    },
+
+    log: function(x, base) {
+        if (base == null) {
+            return new batavia.types.Float(Math.log(x.__float__().val));
+        }
+        return new batavia.types.Float(Math.log(x.__float__().val) / Math.log(base.__float__().val));
+    },
+
+    // TODO: this isn't supported in PhantomJS
+    log10: function(x) {
+        return batavia.modules.math.log10(x);
```

Comment:
Am I misreading this, or is this a recursive call? Shouldn't this be `batavia.modules.math.log(x)`